### PR TITLE
iOS repository_rule example / reorg examples

### PR DESCRIPTION
**Pull request by jerrymarino**
This PR adds an iOS application example using the `repository_rule`

This PR both makes it easier to iterate on examples, and define what is
built in an example with Bazel - just edit Examples/*/BUILD to include
the targets.

Previously, examples were hidden in the `BuildTests` directory, and
undiscoverable. Additionally, `BuildTests` did some interesting things,
which made it hard to manually develop and iterate on Examples.

`BuildTests` is now a very lightweight wrapper around `make/bazel`

Future work:
- due to building of `RepoTools` the rule isn't 100% useable yet
- support `vendorizing` deps where `rules_pods` is in the `WORKSPACE`

### CODE REVIEW

File: Examples/BasiciOS/tools/bazelwrapper [hunk 1]
```diff
@@ -0,0 +1 @@
+../../../tools/bazelwrapper
```
Comment (jerrymarino) [inline Examples/BasiciOS/tools/bazelwrapper]: This is uneeded due to recent additions to `Example.Makefile`

File: Tests/BuildTests/BuildTest.swift [hunk 2]
```diff
@@ -34,8 +34,8 @@ class BuildTests: XCTestCase {
             }
             let bazelScript = "make -C \(rootDir)/Examples/\(example)"
             print("running bazel:", bazelScript)
-            let buildResult = ShellTask(command: "/bin/bash", arguments: [
-                    "-c", bazelScript ], timeout: 1200.0).launch()
+            let buildResult = ShellTask(command: "/bin/bash", arguments: [ "-c",
+                    bazelScript], timeout: 1200.0, streamOutput: true).launch()
             XCTAssertEqual(buildResult.terminationStatus, 0, "building \(example)")
             print("Bazel completed with stderror:", buildResult.standardErrorAsString)
```
Comment (jerrymarino) [inline Tests/BuildTests/BuildTest.swift]: If we stream output here, these prints are probably not needed.

File: Examples/BasiciOS/main.m [hunk 3]
```diff
@@ -0,0 +1,3 @@
+int main(int argc, char * argv[]) {
+    return 0;
```
Comment (mikezucc) [inline Examples/BasiciOS/main.m]: reference symbols from the SLKTextViewController as sanity check?

Comment (jerrymarino) [reply Examples/BasiciOS/main.m]: @mikezucc good suggestion 👍

**Pull request merged**